x86_64/src/arch/devcons.rs

```diff
@@ -0,0 +1,35 @@
+// Racy to start.
+
+use core::fmt;
+use port::devcons::{Console, Uart};
+
+struct Uart16550 {
+    port: u16,
+}
+
+impl Uart for Uart16550 {
+    fn putb(&self, b: u8) {
+        crate::x86_64::uart16550::putb(self.port, b);
+    }
+}
+
+// It would be nice if most the below code was in port....
+
+pub fn print(args: fmt::Arguments) {
+    use core::fmt::Write;
+    let mut cons = Console::new(Uart16550 { port: 0x3f8 });
```

Comment:
It looks like this line is the only thing keeping this code, and println!()/print!(), in x86_64. No matter, let's fix that up in a follow-on.